src/providers/info.rs

```diff
@@ -345,6 +347,50 @@ impl DefaultInfoProvider<'_> {
     }
 }
 
+async fn safe_app_info_from_safe_config_service(
+    url: &str,
+    client: &Arc<dyn HttpClient>,
+    cache: &Arc<dyn Cache>,
+) -> ApiResult<SafeAppInfo> {
+    let config_service_url = config_uri!("/v1/safe-apps/?url={}", url);
+
+    let result = RequestCached::new(config_service_url, client, cache)
+        .execute()
+        .await?;
+
+    let config_safe_apps: Vec<SafeApp> = serde_json::from_str::<Vec<SafeApp>>(&result)?;
+
+    match config_safe_apps.first() {
+        None => {
+            bail!("No Safe Apps match the url")
+        }
+        Some(first) => Ok(SafeAppInfo::from(first)),
+    }
```

Comment:
One issue with this is that if for some reason the query parameter is done in a wrong way, all safe-apps are returned since this is just a filter on the Safe Apps collection.

According to our model, multiple apps can also have the same url. If we expect a single URL to match a single Safe App then we can assert that we get a single result back (but we will have the same issue if the total collection of safe apps with no filters is 1)

Reply:
This works for know. What was the reasoning of letting multiple apps have the same url?

Reply:
> This works for know. What was the reasoning of letting multiple apps have the same url?

I don't think there's a strong reason for it (as in it was never discussed). But we can align on this and have the `url` as a unique field (probably wouldn't work with `http://example.com` vs `http://example.com/` but we already have that issue when querying)

Reply:
I faced that problem some time ago... it was very tricky, at last I used something like https://pypi.org/project/url-normalize/